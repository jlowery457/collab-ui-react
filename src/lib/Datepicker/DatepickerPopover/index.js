import React from 'react';
import PropTypes from 'prop-types';

export default class DatepickerPopover extends React.PureComponent {
  static displayName = 'DatepickerPopover';

  render() {
    const { children } = this.props;

    return (
      /* eslint-disable */
      // Disabled due to onClick on div - needed to not close Popover Prematurely
      <div
        className={'cui-datepicker__popover-container'}
        onClick={e => e.stopPropagation()}>
        <div className={'cui-datepicker__popover'}>
          <div className="inline-flex">{children}</div>
        </div>
      </div>
      /* eslint-enable */
    );
  }
}

DatepickerPopover.defaultProps = {
  children: null,
};

DatepickerPopover.propTypes = {
  /** Child component to display within Popover */
  children: PropTypes.node,
};
