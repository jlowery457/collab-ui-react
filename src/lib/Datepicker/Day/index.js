import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import * as utils from '@collab-ui/react/Datepicker/date_utils';

export default class Day extends React.Component {
  static displayName = 'Day';

  handleClick = event => {
    if (!this.isDisabled() && this.props.onClick) {
      this.props.onClick(event);
    }
  };

  handleKeyPress = e => {
    const { onClick } = this.props;
    e.preventDefault();
    if (e.which === 32 || e.which === 13) {
      return onClick(e);
    } else if (e.charCode === 32 || e.charCode === 13) {
      return onClick(e);
    }
  };

  handleMouseEnter = event => {
    if (!this.isDisabled() && this.props.onMouseEnter) {
      this.props.onMouseEnter(event);
    }
  };

  isSameDay = other => utils.isSameDay(this.props.day, other);

  isDisabled = () => utils.isDayDisabled(this.props.day, this.props);

  isOutsideMonth = () => {
    return (
      this.props.month !== undefined &&
      this.props.month !== utils.getMonth(this.props.day)
    );
  };

  isKeyboardSelected = () =>
    !this.isSameDay(this.props.selected) &&
    this.isSameDay(this.props.preSelection);

  getClassNames = () => {
    return classnames(
      'cui-datepicker__day',

      {
        'cui-datepicker__day--disabled': this.isDisabled(),
        'cui-datepicker__day--selected': this.isSameDay(this.props.selected),
        'cui-datepicker__day--keyboard-selected': this.isKeyboardSelected() && !this.isSameDay(utils.newDate()),
        'cui-datepicker__day--today': this.isSameDay(utils.newDate()) && !this.props.selected,
        'cui-datepicker__day--today--not-selected': this.props.selected && this.isSameDay(utils.newDate()) && !this.isSameDay(this.props.selected),
        'cui-datepicker__day--outside-month': this.isOutsideMonth()
      }
    );
  };

  render() {
    return (
      <div className='cui-datepicker__day-container'>
        <div
          className={this.getClassNames()}
          onClick={this.handleClick}
          onKeyPress={this.handleKeyPress}
          onMouseEnter={this.handleMouseEnter}
          aria-label={`day-${utils.getDate(this.props.day)}`}
          aria-selected={this.isSameDay(this.props.selected)}
          tabIndex='0' //TODO: (jlowery) not sure what to do with this.
          role='option'
        >
          {utils.getDate(this.props.day)}
        </div>
      </div>
    );
  }
}

Day.propTypes = {
  day: PropTypes.object.isRequired,
  month: PropTypes.number,
  onClick: PropTypes.func,
  onMouseEnter: PropTypes.func,
  preSelection: PropTypes.object,
  selected: PropTypes.object,
};