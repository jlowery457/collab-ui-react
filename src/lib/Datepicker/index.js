/**
 * @category controls
 * @component date-picker
 * @variations collab-ui-react
 */

import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import Icon from '@collab-ui/react/Icon';
import Input from '@collab-ui/react/Input';
import DatepickerPopover from '@collab-ui/react/Datepicker/DatepickerPopover';
import Calendar from '@collab-ui/react/Datepicker/Calendar';
import EventOverlay from '@collab-ui/react/EventOverlay';
import * as utils from '@collab-ui/react/Datepicker/date_utils';

export default class DatePicker extends React.Component {
  static displayName = 'DatePicker';

  constructor(props) {
    super(props);
    this.state = this.calcInitialState();
  }

  componentWillReceiveProps(nextProps) {
    const currentMonth = this.props.selected && utils.getMonth(this.props.selected);
    const nextMonth = nextProps.selected && utils.getMonth(nextProps.selected);
    if (currentMonth !== nextMonth) {
      this.setPreSelection(nextProps.selected);
    }
  }

  calcInitialState = () => {
    const defaultPreSelection = utils.now();
    const minDate = utils.getEffectiveMinDate(this.props);
    const maxDate = utils.getEffectiveMaxDate(this.props);
    const boundedPreSelection =
      minDate && utils.isBefore(defaultPreSelection, minDate)
        ? minDate
        : maxDate && utils.isAfter(defaultPreSelection, maxDate)
          ? maxDate
          : defaultPreSelection;
    return {
      activeIndex: null,
      anchorNode: null,
      anchorWidth: null,
      isOpen: false,
      preSelection: this.props.selected
        ? utils.newDate(this.props.selected)
        : boundedPreSelection,
      visibleProp: 'bottom-center',
    };
  };

  choosePosition = () => {
    this.state.isOpen
      && this.isVisible(ReactDOM.findDOMNode(this.calendar), this.state.anchorNode);
  };

  setOpen = open => {
    this.setState({
      isOpen: open,
      preSelection:
        open && this.state.isOpen
          ? this.state.preSelection
          : this.calcInitialState().preSelection,
      anchorNode:
        open
          ? ReactDOM.findDOMNode(this.clickTextField).parentNode
          : null
    }, () => {
      if (open) {
        this.choosePosition();
      }
    });
  };

  isVisible = (element, elementAnchor) => {
    let tempParentArr = [];
    const anchor = elementAnchor && elementAnchor.getBoundingClientRect();
    const elementBoundingRect = element.getBoundingClientRect();
    const elementParent = element.parentElement;
    const windowBottom = window.pageXOffset + window.innerHeight;
    const elementHeight = elementBoundingRect.height;
    const anchorWidth = anchor.width;
    const anchorBottom = anchor.bottom;
    const totalHeight = anchorBottom + elementHeight;

    const findParents = elem => {
      return !elem.parentElement
        ? tempParentArr
        : findParents(elem.parentElement, tempParentArr.push(elem));
    };

    const elementParents = findParents(elementParent);

    const findOverflow = node => {
      const searchProps = ['overflow', 'overflow-y'];

      return searchProps.reduce((agg, prop) => {
        let overflowElement = ReactDOM.findDOMNode(node).style[prop];

        return !overflowElement || agg.includes(overflowElement)
          ? agg
          : (agg += overflowElement);
      }, '');
    };

    const findScrollParent = () => {
      let overflowElement = null;
      let idx = 0;

      while (!overflowElement && elementParents[idx]) {
        if (/(auto|scroll)/.test(findOverflow(elementParents[idx]))) {
          return (overflowElement = elementParents[idx]);
        }
        idx++;
      }

      return overflowElement ? overflowElement : window;
    };

    const scrollParent = findScrollParent(element);
    const parentBottom =
      (!!scrollParent.getBoundingClientRect &&
        scrollParent.getBoundingClientRect().bottom) ||
      windowBottom;

    return (
      (totalHeight < parentBottom && totalHeight < windowBottom)
        ? this.setState({ visibleProp: 'bottom-center', anchorWidth })
        : this.setState({ visibleProp: 'top-center', anchorWidth })
    );
  };

  // onFocus = () => {
  //   this.setState({
  //     isOpen: true,
  //     anchorNode: ReactDOM.findDOMNode(this.clickTextField).parentNode
  //   });
  // };

  onMouseDown = () => {
    if (!this.props.disabled) {
      this.setOpen(true);
    }
  };

  onOverlayClose = () => {
    this.setOpen(false);
  }

  handleChange = event => {
    this.setState({ inputValue: event.target.value });
    const date = utils.parseDate(event.target.value, this.props);
    if (date || !event.target.value) {
      this.setSelected(date, event, true);
    }
  };

  handleSelect = (date, event) => {
    this.setSelected(date, event);
    if (!this.props.shouldCloseOnSelect) {
      this.setPreSelection(date);
    } else {
      this.setOpen(false);
    }
  };

  setSelected = (date, event, keepInput) => {
    let changedDate = date;

    if (changedDate !== null && utils.isDayDisabled(changedDate, this.props)) {
      return;
    }

    if (
      !utils.isSameDay(this.props.selected, changedDate) ||
      this.props.allowSameDay
    ) {
      if (changedDate !== null) {
        if (this.props.selected) {
          let selected = this.props.selected;
          if (keepInput) {
            selected = utils.newDate(changedDate);
          }
          changedDate = utils.setTime(utils.newDate(changedDate), {
            hour: utils.getHour(selected),
            minute: utils.getMinute(selected),
            second: utils.getSecond(selected)
          });
        }
        this.setState({
          preSelection: changedDate
        });
      }
      this.props.onChange(changedDate, event);
    }

    this.props.onSelect(changedDate, event);

    if (!keepInput) {
      this.setState({ inputValue: null });
    }
  };

  setPreSelection = date => {
    const isDateRangePresent =
      typeof this.props.minDate !== 'undefined' &&
      typeof this.props.maxDate !== 'undefined';
    const isValidDateSelection =
      isDateRangePresent && date
        ? utils.isDayInRange(date, this.props.minDate, this.props.maxDate)
        : true;
    if (isValidDateSelection) {
      this.setState({
        preSelection: date
      });
    }
  };

  onInputClick = () => {
    if (!this.props.disabled) {
      this.setOpen(true);
    }
  };

  onInputKeyDown = event => {
    this.props.onKeyDown(event);
    const eventKey = event.key;
    if (!this.state.isOpen) {
      if (eventKey !== 'Enter' && eventKey !== 'Escape' && eventKey !== 'Tab') {
        this.onInputClick();
      }
      return;
    }
    const copy = utils.newDate(this.state.preSelection);
    if (eventKey === 'Enter') {
      event.preventDefault();
      if (
        utils.isMoment(this.state.preSelection) ||
        utils.isDate(this.state.preSelection)
      ) {
        this.handleSelect(copy, event);
        !this.props.shouldCloseOnSelect && this.setPreSelection(copy);
      } else {
        this.setOpen(false);
      }
    } else if (eventKey === 'Escape') {
      event.preventDefault();
      this.setOpen(false);
    } else if (eventKey === 'Tab') {
      this.setOpen(false);
    } else {
      let newSelection;
      switch (eventKey) {
        case 'ArrowLeft':
          event.preventDefault();
          newSelection = utils.subtractDays(copy, 1);
          break;
        case 'ArrowRight':
          event.preventDefault();
          newSelection = utils.addDays(copy, 1);
          break;
        case 'ArrowUp':
          event.preventDefault();
          newSelection = utils.subtractWeeks(copy, 1);
          break;
        case 'ArrowDown':
          event.preventDefault();
          newSelection = utils.addWeeks(copy, 1);
          break;
        case 'PageUp':
          event.preventDefault();
          newSelection = utils.subtractMonths(copy, 1);
          break;
        case 'PageDown':
          event.preventDefault();
          newSelection = utils.addMonths(copy, 1);
          break;
        case 'Home':
          event.preventDefault();
          newSelection = utils.subtractYears(copy, 1);
          break;
        case 'End':
          event.preventDefault();
          newSelection = utils.addYears(copy, 1);
          break;
      }
      // if (this.props.adjustDateOnChange) {
      //   this.setSelected(newSelection);
      // }
      this.setPreSelection(newSelection);
    }
  }

  renderCalendar = () => {
    return (
      <Calendar
        dateFormat={this.props.dateFormatCalendar}
        excludeDates={this.props.excludeDates}
        filterDate={this.props.filterDate}
        includeDates={this.props.includeDates}
        locale={this.props.locale}
        maxDate={this.props.maxDate}
        minDate={this.props.minDate}
        onMonthChange={this.props.onMonthChange}
        onSelect={this.handleSelect}
        preSelection={this.state.preSelection}
        selected={this.props.selected}
      />
    );
  }

  renderInput = () => {
    const inputValue =
      typeof this.props.value === 'string'
        ? this.props.value
        : typeof this.state.inputValue === 'string'
          ? this.state.inputValue
          : utils.safeDateFormat(this.props.selected, this.props);

    return (
      <Input
        disabled={this.props.disabled}
        htmlId={`${this.props.htmlId}-input`}
        inputRef={ref => (this.clickTextField = ref)}
        inputSize={this.props.inputSize}
        label={this.props.label}
        name={`${this.props.name}-input`}
        onChange={this.handleChange}
        onClick={this.onInputClick}
        onKeyDown={this.onInputKeyDown}
        onMouseDown={this.onMouseDown}
        ref={ref => this.calendar = ref}
        // onFocus={this.onFocus}
        placeholder={this.props.placeholder}
        value={inputValue}
      >
        <Icon name='calendar-month_20' isAria={false} />
      </Input>
    );
  }

  render() {
    const calendar = this.renderCalendar();
    this.input = this.renderInput();

    const popoverElement = (
      <EventOverlay
        allowClickAway
        anchorNode={this.state.anchorNode}
        close={this.onOverlayClose}
        direction={this.state.visibleProp}
        isOpen={this.state.isOpen}>
        <DatepickerPopover>
          {calendar}
        </DatepickerPopover>
      </EventOverlay>
    );

    return (
      <div className='cui-datepicker-container'>
        {this.input}
        {popoverElement}
      </div>
    );
  }
}

DatePicker.propTypes = {
  allowSameDay: PropTypes.bool,
  dateFormat: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
  dateFormatCalendar: PropTypes.string,
  disabled: PropTypes.bool,
  excludeDates: PropTypes.array,
  filterDate: PropTypes.func,
  htmlId: PropTypes.string,
  includeDates: PropTypes.array,
  inputSize: PropTypes.string,
  label: PropTypes.string,
  locale: PropTypes.string,
  maxDate: PropTypes.object,
  minDate: PropTypes.object,
  name: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onKeyDown: PropTypes.func,
  onMonthChange: PropTypes.func,
  placeholder: PropTypes.string,
  onSelect: PropTypes.func,
  selected: PropTypes.object,
  shouldCloseOnSelect: PropTypes.bool,
  value: PropTypes.string,
};

DatePicker.defaultProps = {
  allowSameDay: false,
  dateFormat: 'L',
  dateFormatCalendar: 'MMMM YYYY',
  onChange() {},
  onKeyDown() {},
  onMonthChange() {},
  onSelect() {},
  shouldCloseOnSelect: true,
};
