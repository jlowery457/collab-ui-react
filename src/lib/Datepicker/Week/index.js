import React from 'react';
import PropTypes from 'prop-types';
import Day from '@collab-ui/react/Datepicker/Day';
import * as utils from '@collab-ui/react/Datepicker/date_utils';

export default class Week extends React.Component {
  static displayName = 'Week';

  handleDayClick = (day, event) => {
    if (this.props.onDayClick) {
      this.props.onDayClick(day, event);
    }
  };

  handleDayMouseEnter = day => {
    if (this.props.onDayMouseEnter) {
      this.props.onDayMouseEnter(day);
    }
  };

  renderDays = () => {
    const startOfWeek = utils.getStartOfWeek(utils.cloneDate(this.props.day));
    const days = [];

    return days.concat(
      [0, 1, 2, 3, 4, 5, 6].map(offset => {
        const day = utils.addDays(utils.cloneDate(startOfWeek), offset);
        return (
          <Day
            key={offset}
            day={day}
            excludeDates={this.props.excludeDates}
            filterDate={this.props.filterDate}
            includeDates={this.props.includeDates}
            minDate={this.props.minDate}
            maxDate={this.props.maxDate}
            month={this.props.month}
            onClick={this.handleDayClick.bind(this, day)}
            onMouseEnter={this.handleDayMouseEnter.bind(this, day)}
            preSelection={this.props.preSelection}
            selected={this.props.selected}
          />
        );
      })
    );
  };

  render() {
    return <div className="cui-datepicker__week">{this.renderDays()}</div>;
  }
}

Week.propTypes = {
  day: PropTypes.object.isRequired,
  excludeDates: PropTypes.array,
  filterDate: PropTypes.func,
  includeDates: PropTypes.array,
  maxDate: PropTypes.object,
  minDate: PropTypes.object,
  month: PropTypes.number,
  onDayClick: PropTypes.func,
  onDayMouseEnter: PropTypes.func,
  preSelection: PropTypes.object,
  selected: PropTypes.object,
};