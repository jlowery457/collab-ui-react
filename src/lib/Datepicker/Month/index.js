import React from 'react';
import PropTypes from 'prop-types';
import Week from '@collab-ui/react/Datepicker/Week';
import * as utils from '@collab-ui/react/Datepicker/date_utils';

const FIXED_HEIGHT_STANDARD_WEEK_COUNT = 6;

export default class Month extends React.Component {
  static displayName = 'Month';

  handleDayClick = (day, event) => {
    if (this.props.onDayClick) {
      this.props.onDayClick(day, event);
    }
  };

  isWeekInMonth = startOfWeek => {
    const day = this.props.day;
    const endOfWeek = utils.addDays(utils.cloneDate(startOfWeek), 6);
    return (
      utils.isSameMonth(startOfWeek, day) || utils.isSameMonth(endOfWeek, day)
    );
  };

  renderWeeks = () => {
    const weeks = [];
    const isFixedHeight = this.props.fixedHeight;
    let currentWeekStart = utils.getStartOfWeek(
      utils.getStartOfMonth(utils.cloneDate(this.props.day))
    );
    let i = 0;
    let breakAfterNextPush = false;

    while (true) {
      weeks.push(
        <Week
          key={i}
          day={currentWeekStart}
          excludeDates={this.props.excludeDates}
          filterDate={this.props.filterDate}
          includeDates={this.props.includeDates}
          maxDate={this.props.maxDate}
          minDate={this.props.minDate}
          month={utils.getMonth(this.props.day)}
          onDayClick={this.handleDayClick}
          preSelection={this.props.preSelection}
          selected={this.props.selected}
        />
      );

      if (breakAfterNextPush) {
        break;
      } 

      i++;
      currentWeekStart = utils.addWeeks(utils.cloneDate(currentWeekStart), 1);

      // If one of these conditions is true, we will either break on this week
      // or break on the next week
      const isFixedAndFinalWeek =
        isFixedHeight && i >= FIXED_HEIGHT_STANDARD_WEEK_COUNT;
      const isNonFixedAndOutOfMonth =
        !isFixedHeight && !this.isWeekInMonth(currentWeekStart);

      if (isFixedAndFinalWeek || isNonFixedAndOutOfMonth) {
        if (this.props.peekNextMonth) {
          breakAfterNextPush = true;
        } else {
          break;
        }
      }
    }

    return weeks;
  };


  render() {
    return (
      <div
        className='cui-datepicker__month'
        onMouseLeave={this.handleMouseLeave}
        // role="listbox"
      >
        {this.renderWeeks()}
      </div>
    );
  }
}

Month.propTypes = {
  day: PropTypes.object.isRequired,
  excludeDates: PropTypes.array,
  filterDate: PropTypes.func,
  fixedHeight: PropTypes.bool,
  includeDates: PropTypes.array,
  maxDate: PropTypes.object,
  minDate: PropTypes.object,
  onDayClick: PropTypes.func,
  peekNextMonth: PropTypes.bool,
  preSelection: PropTypes.object,
  selected: PropTypes.object,
};
