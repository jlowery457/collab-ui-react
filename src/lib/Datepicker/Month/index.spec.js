import React from 'react';
import { shallow, mount } from 'enzyme';
import Month from '../Month';
import Week from '@collab-ui/react/Datepicker/Week';
import Day from '@collab-ui/react/Datepicker/Day';
import * as utils from '@collab-ui/react/Datepicker/date_utils';
import moment from 'moment';
import range from 'lodash/range';

const TEST_DATE = '2018-04-01';

describe('tests for <Month />', () => {
  const day = moment(TEST_DATE);

  expect.extend({
    toBeSameDay(received, argument) {
      const pass = utils.isSameDay(received, argument);
      if (pass) {
        return { 
          message: () => 
            `expected ${received} not to be equal to ${argument}`,
          pass: true,
        };
      } else {
        return {
          message: () => `expected ${received} to be equal to ${argument}`,
          pass: false,
        };
      }
    }
  });

  function assertDateRangeInclusive(month, start, end) {
    const dayCount = utils.getDaysDiff(end, start) + 1;
    const days = month.find(Day);
    expect(days).toHaveLength(dayCount);
    range(0, dayCount).forEach(offset => {
      const day = days.get(offset);
      const expectedDay = utils.addDays(utils.cloneDate(start), offset);
      expect(day.props.day).toBeSameDay(expectedDay);
    });
  }

  it('should match month SnapShot', () => {
    const container = shallow(
      <Month day={day} month={3} />
    );

    expect(container).toMatchSnapshot();
  });

  it('should render 5 Week Components', () => {
    const container = shallow(
      <Month day={day} month={3} />
    );

    expect(container.find(Week)).toHaveLength(5);
  });

  it('should render all days of the month and some days in neighboring months', () => {
    assertDateRangeInclusive(
      mount(<Month day={day} />),
      utils.getStartOfWeek(utils.cloneDate(day)),
      utils.getEndOfWeek(utils.getEndOfMonth(utils.cloneDate(day)))
    );
  });

  it('should handle day click event', () => {
    let count = 0;
    const container = mount(
      <Month day={day} month={3} onDayClick={() => count++} />
    );

    container.find(Day).first().find('div.cui-datepicker__day').simulate('click');
    expect(count).toEqual(1);
  });

});
