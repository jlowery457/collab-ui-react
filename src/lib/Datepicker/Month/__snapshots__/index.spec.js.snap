// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for <Month /> should match month SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Month
    day={"2018-04-01T05:00:00.000Z"}
    month={3}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Week
          day={"2018-04-01T05:00:00.000Z"}
          excludeDates={undefined}
          filterDate={undefined}
          includeDates={undefined}
          maxDate={undefined}
          minDate={undefined}
          month={3}
          onDayClick={[Function]}
          preSelection={undefined}
          selected={undefined}
        />,
        <Week
          day={"2018-04-08T05:00:00.000Z"}
          excludeDates={undefined}
          filterDate={undefined}
          includeDates={undefined}
          maxDate={undefined}
          minDate={undefined}
          month={3}
          onDayClick={[Function]}
          preSelection={undefined}
          selected={undefined}
        />,
        <Week
          day={"2018-04-15T05:00:00.000Z"}
          excludeDates={undefined}
          filterDate={undefined}
          includeDates={undefined}
          maxDate={undefined}
          minDate={undefined}
          month={3}
          onDayClick={[Function]}
          preSelection={undefined}
          selected={undefined}
        />,
        <Week
          day={"2018-04-22T05:00:00.000Z"}
          excludeDates={undefined}
          filterDate={undefined}
          includeDates={undefined}
          maxDate={undefined}
          minDate={undefined}
          month={3}
          onDayClick={[Function]}
          preSelection={undefined}
          selected={undefined}
        />,
        <Week
          day={"2018-04-29T05:00:00.000Z"}
          excludeDates={undefined}
          filterDate={undefined}
          includeDates={undefined}
          maxDate={undefined}
          minDate={undefined}
          month={3}
          onDayClick={[Function]}
          preSelection={undefined}
          selected={undefined}
        />,
      ],
      "className": "cui-datepicker__month",
      "onMouseLeave": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "day": "2018-04-01T05:00:00.000Z",
          "excludeDates": undefined,
          "filterDate": undefined,
          "includeDates": undefined,
          "maxDate": undefined,
          "minDate": undefined,
          "month": 3,
          "onDayClick": [Function],
          "preSelection": undefined,
          "selected": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "day": "2018-04-08T05:00:00.000Z",
          "excludeDates": undefined,
          "filterDate": undefined,
          "includeDates": undefined,
          "maxDate": undefined,
          "minDate": undefined,
          "month": 3,
          "onDayClick": [Function],
          "preSelection": undefined,
          "selected": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "day": "2018-04-15T05:00:00.000Z",
          "excludeDates": undefined,
          "filterDate": undefined,
          "includeDates": undefined,
          "maxDate": undefined,
          "minDate": undefined,
          "month": 3,
          "onDayClick": [Function],
          "preSelection": undefined,
          "selected": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "class",
        "props": Object {
          "day": "2018-04-22T05:00:00.000Z",
          "excludeDates": undefined,
          "filterDate": undefined,
          "includeDates": undefined,
          "maxDate": undefined,
          "minDate": undefined,
          "month": 3,
          "onDayClick": [Function],
          "preSelection": undefined,
          "selected": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "class",
        "props": Object {
          "day": "2018-04-29T05:00:00.000Z",
          "excludeDates": undefined,
          "filterDate": undefined,
          "includeDates": undefined,
          "maxDate": undefined,
          "minDate": undefined,
          "month": 3,
          "onDayClick": [Function],
          "preSelection": undefined,
          "selected": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Week
            day={"2018-04-01T05:00:00.000Z"}
            excludeDates={undefined}
            filterDate={undefined}
            includeDates={undefined}
            maxDate={undefined}
            minDate={undefined}
            month={3}
            onDayClick={[Function]}
            preSelection={undefined}
            selected={undefined}
          />,
          <Week
            day={"2018-04-08T05:00:00.000Z"}
            excludeDates={undefined}
            filterDate={undefined}
            includeDates={undefined}
            maxDate={undefined}
            minDate={undefined}
            month={3}
            onDayClick={[Function]}
            preSelection={undefined}
            selected={undefined}
          />,
          <Week
            day={"2018-04-15T05:00:00.000Z"}
            excludeDates={undefined}
            filterDate={undefined}
            includeDates={undefined}
            maxDate={undefined}
            minDate={undefined}
            month={3}
            onDayClick={[Function]}
            preSelection={undefined}
            selected={undefined}
          />,
          <Week
            day={"2018-04-22T05:00:00.000Z"}
            excludeDates={undefined}
            filterDate={undefined}
            includeDates={undefined}
            maxDate={undefined}
            minDate={undefined}
            month={3}
            onDayClick={[Function]}
            preSelection={undefined}
            selected={undefined}
          />,
          <Week
            day={"2018-04-29T05:00:00.000Z"}
            excludeDates={undefined}
            filterDate={undefined}
            includeDates={undefined}
            maxDate={undefined}
            minDate={undefined}
            month={3}
            onDayClick={[Function]}
            preSelection={undefined}
            selected={undefined}
          />,
        ],
        "className": "cui-datepicker__month",
        "onMouseLeave": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "day": "2018-04-01T05:00:00.000Z",
            "excludeDates": undefined,
            "filterDate": undefined,
            "includeDates": undefined,
            "maxDate": undefined,
            "minDate": undefined,
            "month": 3,
            "onDayClick": [Function],
            "preSelection": undefined,
            "selected": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "day": "2018-04-08T05:00:00.000Z",
            "excludeDates": undefined,
            "filterDate": undefined,
            "includeDates": undefined,
            "maxDate": undefined,
            "minDate": undefined,
            "month": 3,
            "onDayClick": [Function],
            "preSelection": undefined,
            "selected": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "day": "2018-04-15T05:00:00.000Z",
            "excludeDates": undefined,
            "filterDate": undefined,
            "includeDates": undefined,
            "maxDate": undefined,
            "minDate": undefined,
            "month": 3,
            "onDayClick": [Function],
            "preSelection": undefined,
            "selected": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "day": "2018-04-22T05:00:00.000Z",
            "excludeDates": undefined,
            "filterDate": undefined,
            "includeDates": undefined,
            "maxDate": undefined,
            "minDate": undefined,
            "month": 3,
            "onDayClick": [Function],
            "preSelection": undefined,
            "selected": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "class",
          "props": Object {
            "day": "2018-04-29T05:00:00.000Z",
            "excludeDates": undefined,
            "filterDate": undefined,
            "includeDates": undefined,
            "maxDate": undefined,
            "minDate": undefined,
            "month": 3,
            "onDayClick": [Function],
            "preSelection": undefined,
            "selected": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
